File: WordSoulApi/WordSoulApi/Models/DTOs/User/UserDto.cs
using System.ComponentModel.DataAnnotations;
using WordSoulApi.Models.Entities;

namespace WordSoulApi.Models.DTOs.User
{
    public class UserDto
    {
        public int Id { get; set; }
        public string Username { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string? Role { get; set; }
        public DateTime CreatedAt { get; set; }
        public bool IsActive { get; set; }
    }

    

    public class UserDetailDto : UserDto
    {
        public int Level { get; set; }
        public int TotalXP { get; set; }
        public int TotalAP { get; set; }
        public int StreakDays { get; set; }
        public int PetCount { get; set; }
        public string? AvatarUrl { get; set; }
    }

   

    public class LeaderBoardDto
    {
        public int UserId { get; set; }
        public string UserName { get; set; } = string.Empty;
        public int totalXP { get; set; }
        public int totalAP { get; set; }
    }


}

File: wordsoul-app/src/types/UserDto.ts
export interface LoginDto {
  username: string;
  password: string;
}

export interface RegisterDto {
  username: string;
  email: string;
  password: string;
}

export interface RefreshTokenRequestDto {
  id: number;
  refreshToken: string;
}

export interface TokenResponseDto {
  accessToken: string;
  refreshToken: string;
}


export interface UserDto {
  id: number;
  username: string;
  email: string;
  role: string;
  createdAt: string;
  isActive: boolean;
  level: number;
  totalXP: number;
  totalAP: number;
  streakDays: number;
  avatarUrl?: string;
  petCount?: number;
}

export interface LevelStatDto {
  level: number;
  count: number;
}

export interface UserProgressDto {
  reviewWordCount: number;
  nextReviewTime: string | null;
  vocabularyStats: LevelStatDto[];
}

export interface ActivityLogDto {
  id: number;
  userId: number;
  username: string;
  action: string;
  details: string;
  timestamp: Date;
}

export interface AssignRoleDto {
  roleName: string;
}

export interface UserVocabularySetDto {
  vocabularySetId: number;
  totalCompletedSessions: number;
  isCompleted: boolean;
  createdAt: string;
  isActive: boolean;
}

export interface LeaderBoardDto {
  userId: number;
  userName: string;
  totalXP: number;
  totalAP: number;
}